### PR TITLE
Add WSL test infrastructure

### PR DESCRIPTION
**Pull request by lucasmoura**
## Why is this needed?
To properly test the Pro client on WSL instances, we need to have an infrastructure that allow us to launch those instance in a Windows machine, install the required Pro client package on them and run the pro commands.

This PR is adding that infrastructure, but there are still some items we need to discuss before merging.

1) Are we going to use a default name for the windows machine or should that be configurable ?
2) We are not using all of the shell safe mechanisms we have for the Ubuntu based tests, should we consider making use of them here too ?
3) Should we mark all container tests as WSL supported ?

## Test Steps
After setting up a windows machine, use the provided infrastructure to run any test marked as supporting WSL

## Checklist

 - [ ] I have updated or added any unit tests accordingly
 - [x] I have updated or added any integration tests accordingly
 - [ ] Changes here need to be documented, and this was done in: 

## Does this PR require extra reviews?

 - [ ] Yes
 - [x] No

### CODE REVIEW

**Review by orndorffgrant — COMMENTED** [review]
Sorry I typed this comment forever ago and just noticed I never submitted :facepalm:

**Review by orndorffgrant — COMMENTED** [review]
Seems to work! Just a clean up error I mention in a comment. After that is resolved I'll test again to make sure the wsl instance is stopped/started correctly before approving

File: features/environment.py [hunk 1]
```diff
@@ -450,6 +460,11 @@ def after_all(context):
                 "Failed to delete instance ssh keys:\n{}".format(str(e))
             )
 
+        if "wsl" in context.pro_config.machine_types:
```
[the remark this answers is not in the document]
Comment (lucasmoura) [reply features/environment.py:463]: Fair enough, I will remove that check and leave the one that just checks if  `wsl` appears on the cloud instances

**Review by orndorffgrant — APPROVED** [review]

**Review by renanrodrigo — APPROVED** [review]
let the Windows fun begin!

**Pull request merged**